### ISSUE TITLE
client should reject both being and using an exit node at the same time

### ISSUE THREAD

**Issue by bradfitz** [tailscale/tailscale#3569]
I don't think we've ever considered whether using an exit node & running (advertising) an exit node at the same time should work.

Maybe it works today? Kinda? On some platforms?

But I don't think it works consistently on all platforms today if it even works on any.

And what should it do? Should the exit node that's running forward out of the exit node that you're using?

What if there's a loop?

etc.

Decide.

**Comment by mayakacz** [comment]
From Slack conversation: shouldn't be able to be both exit node and use exit node. Whichever one you selected last is what it should be doing, and automatically disconnect/disable the other option.
We should probably warn you of this/ make this clear, esp. if you are being used as an exit node.

**Comment by mayakacz** [comment]
> Maybe it works today? Kinda? On some platforms?

If it does, we should be clear where that's true. I don't think we've tested all cases.

In at least some cases, it doesn't work. Hit this today with macOS and it caused unexpected behaviour. Was advertising an exit node, then selected an exit node, and traffic did not work. Did not realize I was still advertising an exit node from something I did earlier.

Proposal: similar to how iOS hotspot/wifi interaction used to(?) work:
* If I am advertising an exit node, and try to use an exit node, then give me a warning that this will stop advertising an exit node, before doing so.
* If I am connected to an exit node, and try to advertise an exit node, same.
* If I am neither using an exit node, or advertising an exit node, then using these features works as they currently do

I think it's preferable to give a warning (before applying change), rather than say, make options not selectable, as I don't necessarily know why those are not available. Selecting the option will still tell me why and let me decide.

**Comment by rweirtbg** [comment]
This does work on Linux (Ubuntu 22.04/20.04) and we desperately need this capability for our IOT application for both our IOT appliance to route its internet traffic through tailscale to bypass the customer LAN and improve reliability and the ability for the appliance to access our customer's LAN to reach other devices.

**Comment by bradfitz** [comment]
@rweirtbg, I'm not exactly following your requirements from that description. Why can't you have the source clients use the ultimate exit node? What do you get from hopping via an intermediate exit node?

From #5710 we're thinking of removing this because there's a bunch of undefined behavior that varies per platform and we don't want to guarantee any behavior that "works" today when we didn't try to make anything work. We don't even know what it's supposed to do vs what it actually does.

Like: in your configuration, which LAN(s) should be accessible? How should DNS route?

**Comment by rweirtbg** [comment]
So in our case we'd like the unit to access the internet through a configured exit node, but also let us access the internal network like #5412.

**Comment by DentonGentry** [comment]
> So in our case we'd like the unit to access the internet through a configured exit node, but also let us access the internal network the the unit is connected to like https://github.com/tailscale/tailscale/issues/5412.

We'll track that on https://github.com/tailscale/tailscale/issues/5412.

On the main topic of this issue: exit nodes cannot themselves use another exit node. We are not onion routing through exit nodes.

**Comment by bradfitz** [comment]
@DentonGentry, then we need another bug to track doing the work to disallow this.

**Comment by DentonGentry** [comment]
I'll open an internal issue, I think we want to do that in the coordination server.

**Comment by DentonGentry** [comment]
Fixed in 1.36 by https://github.com/tailscale/tailscale/pull/6383
```
# tailscale up --advertise-exit-node=true --exit-node=hamburg
Cannot advertise an exit node and use an exit node at the same time.
# 
```
